### PR TITLE
Add more statistics

### PR DESCRIPTION
**Pull request by ks129**
Added following new statistics:
- Guild boost statistics
- Markdown code block correction
- `eval` fails and successes
- Who runs `eval`? `Helper` or `Developer`?
- Where `eval` run? Topical, help channels or bot commands

There is no public issue, but one private organization issue.

### CODE REVIEW

**Comment by jos-b** [comment]
Relevant organisation issue: python-discord/organisation#232

**Comment by ks129** [comment]
@jos-b Made fixes

File: bot/cogs/snekbox.py [hunk 1]
```diff
@@ -205,6 +205,12 @@ async def send_eval(self, ctx: Context, code: str) -> Message:
             if paste_link:
                 msg = f"{msg}\nFull output: {paste_link}"
 
+            # Collect stats of eval fails + successes
+            if icon == ":x:":
+                self.bot.stats.incr("snekbox.python.fail")
+            elif icon in (":warning:", ":white_check_mark:"):
```
Comment (MarkKoz) [inline bot/cogs/snekbox.py:211]: Safe to simplify this to an `else`.
```suggestion
            else:
```

File: bot/cogs/snekbox.py [hunk 2]
```diff
@@ -292,6 +298,18 @@ async def eval_command(self, ctx: Context, *, code: str = None) -> None:
             await ctx.invoke(self.bot.get_command("help"), "eval")
             return
 
+        if Roles.helpers in (role.id for role in ctx.author.roles):
+            self.bot.stats.incr("snekbox_usages.roles.helpers")
+        else:
+            self.bot.stats.incr("snekbox_usages.roles.developers")
+
+        if ctx.channel.category_id == Categories.help_in_use:
```
Comment (MarkKoz) [inline bot/cogs/snekbox.py:306]: We don't allow this command in DMs, right? Cause if `channel` is a `DMChannel`, then `category_id` will be an `AttributeError`.

I want to note I have a PR which adds a utility function for checking help channels. It's not done yet, but when it is, this should be replaced with it. https://github.com/python-discord/bot/blob/4378e1f12907a73a26e56760c4a5717ff2a017d2/bot/utils/channel.py#L10

Comment (ks129) [reply bot/cogs/snekbox.py:306]: Yes, this have guild only decorator.

**Pull request merged**